internal/command/meta_backend.go

```diff
@@ -1376,17 +1477,26 @@ hasn't changed and try again. At this point, no changes to your existing
 configuration or state have been made.
 `
 
+const errBackendInitCloudMigration = `
+Reason: %s.
+
+Migrating to Terraform Cloud requires reinitialization, to discover which Terraform Cloud workspaces belong to this configuration and to optionally migrate existing state to the corresponding Terraform Cloud workspaces.
+
+To re-initialize, run:
+  terraform init
+
+Terraform has not yet made changes to your existing configuration or state.
+`
+
 const errBackendInitCloud = `
-Reason: %s
+Reason: %s.
```

Comment:
This was an "while I'm in the area" error message improvement: diagnostic detail messages are always full paragraphs with punctuation, and this was only missing its punctuation because it was originally (prior to v0.12) not a diagnostic message at all.

The motivation here is the same as removing the hard line breaks in some of the other messages: gradually making these old codepaths more consistent with our modern expectations while we are working in the area anyway.